### ISSUE TITLE
Workflow additional preconditions

### ISSUE THREAD

**Issue by mariusz-jachimowicz-83** [onyx-platform/onyx#625]
- When workflow is unbound var 
```clojure
(def workflow)
```
then I have 
```
java.lang.IllegalArgumentException: Don't know how to create ISeq from: clojure.lang.Var$Unbound
```

- when workflow is empty array
```clojure
(def workflow [])
```
then platform is starting but it's hanging.

We need to add additional workflow validation preconditions.

**Comment by MichaelDrogalis** [comment]
A fix for the second case is appropriate, but the error message on the first is plenty clear about the var being unbound.

**Comment by mariusz-jachimowicz-83** [comment]
With my modification I can have for instance
```clojure
#object[clojure.lang.Var$Unbound 0x1a2cbe42 "Unbound: #'onyx.peer.people-flow-test/workflow"]
^-- Value must be of type clojure.lang.PersistentVector (found clojure.lang.Var$Unbound)
```
instead of large stacktrace.

**Comment by MichaelDrogalis** [comment]
There's more to think about than getting a more concise error message here. If we carefully check that the workflow var is unbound, should we also check that the var's representing values for the catalog, lifecycles, triggers, windows, flow conditions, and task scheduler aren't unbound either? And what about values coming in through the peer configuration? The surface area is as high as Clojure itself.

Secondly, the stacktrace is probably more helpful than simply seeing (found "clojure.lang.Var$Unbound"). At what point in time was it found to be unbound? Having a stacktrace lets you work backward. These concise error messages via the analyzer are meant to diagnose semantic problems. An unbound var doesn't fall into that category.

**Comment by lbradstreet** [comment]
As a side note, we probably shouldn't have an unbound var in the default
learn-onyx workshop lessons. I'd personally rather the lesson started with
an empty vector, because I have seen people work through these lessons and
hit this problem and be a little confused.

**Comment by mariusz-jachimowicz-83** [comment]
I removed checking for unbound var. It's preventing workflow empty vector only. Please review and merge if it's ok.

**Comment by MichaelDrogalis** [comment]
Thanks for the submission and bug report @mariusz-jachimowicz-83. I ended up applying a slightly different fix in https://github.com/onyx-platform/onyx/commit/71bd7b9d76d3a42f68182d9c3f9cc2987c066c78 that doesn't create a new semantic error type. I added you to the contributor list in https://github.com/onyx-platform/onyx/commit/883df1f82f669ad8bb42bd82dec43be4234908dd for the submission since the fixes are quite similar. Thanks again for mentioning the problem and submitting a patch, we quite appreciate it!
